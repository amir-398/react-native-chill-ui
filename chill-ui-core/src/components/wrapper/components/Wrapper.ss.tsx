import type { WrapperPropsSs } from '@types';
import type { PropsWithChildren } from 'react';

import { BoxSs } from '@components/box';

import { wrapperSv } from '../styles/Wrapper.ss.styles';
import { WrapperSafeAreaView } from './WrapperSafeAreaView';

/**
 * The `<Wrapper />` component provides a flexible container with default styling and SafeAreaView support.
 *
 *
 * <!-- STORYBOOK_IMPORT_START
 * ```tsx
 * import { Wrapper } from 'react-native-chill-ui';
 * ```
 * STORYBOOK_IMPORT_END -->
 *
 * @example
 * ```tsx
 * <Wrapper style={{ backgroundColor: '#f0f0f0' }}>
 *   <String>Content</String>
 * </Wrapper>
 * ```
 *
 * @param edges - Safe area edges to apply when hasSafeArea is true: `'top'` | `'right'` | `'bottom'` | `'left'`
 * @param fill - Whether to fill the wrapper
 * @param grow - Whether to grow the wrapper
 * @param hasSafeArea - Whether to wrap content in SafeAreaView
 * @param px - Horizontal padding variant: `'none'` | `'xs'` | `'sm'` | `'md'` | `'lg'` | `'xl'`
 * @param ViewProps - Any other props accepted by the native `View` component.
 */
export function Wrapper(props: PropsWithChildren<WrapperPropsSs>) {
  const { children, edges, fill, grow, hasSafeArea, px, style } = props;

  const content = <BoxSs style={[wrapperSv({ fill, grow, px }), style]}>{children}</BoxSs>;

  if (hasSafeArea) {
    return (
      <WrapperSafeAreaView edges={edges} px="none">
        {content}
      </WrapperSafeAreaView>
    );
  }

  return content;
}

Wrapper.displayName = 'Wrapper';
