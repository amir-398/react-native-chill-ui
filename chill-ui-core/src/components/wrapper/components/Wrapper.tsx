import type { WrapperPropsTw } from '@types';
import type { PropsWithChildren } from 'react';

import { Box } from '@components/box';
import { cn, classNamePropsHandler, classNameHandler, styleHandler } from '@utils';

import { wrapperTv } from '../styles/Wrapper.tw.styles';
import { wrapperSv } from '../styles/Wrapper.ss.styles';
import { WrapperSafeAreaView } from './WrapperSafeAreaView';

/**
 * The `<Wrapper />` component provides a flexible container with default styling and SafeAreaView support.
 *
 *
 * <!-- STORYBOOK_IMPORT_START
 * ```tsx
 * import { Wrapper } from 'react-native-chill-ui';
 * ```
 * STORYBOOK_IMPORT_END -->
 *
 * @example
 * ```tsx
 * <Wrapper>
 *   <String>Content</String>
 * </Wrapper>
 * ```
 * @param className - Custom CSS classes for the wrapper (NativeWind only)
 * @param edges - Safe area edges to apply when hasSafeArea is true
 * @param fill - Whether to fill the wrapper
 * @param grow - Whether to grow the wrapper
 * @param hasSafeArea - Whether to wrap content in SafeAreaView
 * @param px - Horizontal padding variant: `'none'` | `'xs'` | `'sm'` | `'md'` | `'lg'` | `'xl'`
 * @param ViewProps - Any other props accepted by the native `View` component.
 */
export function Wrapper(props: PropsWithChildren<WrapperPropsTw>) {
  classNamePropsHandler(props, 'Wrapper');

  const { children, className, edges, fill, grow, hasSafeArea, px, style, ...rest } = props;

  const content = (
    <Box
      {...classNameHandler(cn(wrapperTv({ fill, grow, px }), className))}
      {...styleHandler({ defaultStyle: wrapperSv({ fill, grow, px }), style })}
      {...rest}
    >
      {children}
    </Box>
  );

  if (hasSafeArea) {
    return (
      <WrapperSafeAreaView edges={edges} px="none">
        {content}
      </WrapperSafeAreaView>
    );
  }

  return content;
}

Wrapper.displayName = 'Wrapper';
