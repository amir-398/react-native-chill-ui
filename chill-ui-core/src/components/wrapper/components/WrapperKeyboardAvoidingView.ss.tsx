import type { PropsWithChildren } from 'react';
import type { WrapperKeyboardAvoidingViewPropsSs } from '@types';

import { Wrapper } from './Wrapper.ss';
import { wrapperSv } from '../styles/Wrapper.ss.styles';
import { wrapperDefaultProps } from '../utils/defaultProps';

// Optional import with error handling
let KeyboardAvoidingView: any;

try {
  // eslint-disable-next-line
  const keyboardController = require('react-native-keyboard-controller');
  if (keyboardController) {
    KeyboardAvoidingView = keyboardController.KeyboardAvoidingView;
  }
} catch {
  console.warn(
    'react-native-keyboard-controller is not installed. To use WrapperKeyboardAvoidingView, please install it: npm install react-native-keyboard-controller',
  );
}

/**
 * KeyboardAvoidingView wrapper component for keyboard avoidance.
 *
 * @example
 * ```tsx
 * <WrapperKeyboardAvoidingView behavior="padding">
 *   <Input placeholder="Type here" />
 * </WrapperKeyboardAvoidingView>
 * ```
 *
 * @param fill - Whether to fill the wrapper
 * @param px - Padding for the wrapper
 * @param style - Style prop
 * @param keyboardVerticalOffset - Keyboard vertical offset
 * @param behavior - Behavior of the keyboard avoiding view
 * @param enabled - Whether the keyboard avoiding view is enabled
 * @param contentContainerStyle - Content container style
 * @param children - Child components to render
 */
export function WrapperKeyboardAvoidingView(props: PropsWithChildren<WrapperKeyboardAvoidingViewPropsSs>) {
  const {
    children,
    fill,
    keyboardVerticalOffset = wrapperDefaultProps.keyboardVerticalOffset,
    px,
    style,
    ...rest
  } = props;

  if (!KeyboardAvoidingView) {
    console.error(
      'react-native-keyboard-controller is not installed. To use WrapperKeyboardAvoidingView, please install it: npm install react-native-keyboard-controller',
    );
    return (
      <Wrapper style={[wrapperSv({ fill, px }), style]} {...rest}>
        {children}
      </Wrapper>
    );
  }

  return (
    <KeyboardAvoidingView
      style={[wrapperSv({ fill, px }), style]}
      keyboardVerticalOffset={keyboardVerticalOffset}
      {...rest}
    >
      {children}
    </KeyboardAvoidingView>
  );
}

WrapperKeyboardAvoidingView.displayName = 'WrapperKeyboardAvoidingView';
