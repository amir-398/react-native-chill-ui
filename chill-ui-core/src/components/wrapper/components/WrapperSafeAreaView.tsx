import type { PropsWithChildren } from 'react';
import type { WrapperSafeAreaViewPropsTw } from '@types';

import { classNameHandler, classNamePropsHandler, cn, customConsole, styleHandler } from '@utils';

import { Wrapper } from './Wrapper.tw';
import { wrapperTv } from '../styles/Wrapper.tw.styles';
import { wrapperSv } from '../styles/Wrapper.ss.styles';

// Optional import with error handling
let SafeAreaView: any;

try {
  // eslint-disable-next-line
  const safeAreaContext = require('react-native-safe-area-context');
  if (safeAreaContext) {
    SafeAreaView = safeAreaContext.SafeAreaView;
  }
} catch {
  customConsole.error(
    'react-native-safe-area-context is not installed. To use WrapperSafeAreaView, please install it: npm install react-native-safe-area-context',
  );
}

/**
 * SafeAreaView wrapper component for handling safe areas.
 * Automatically detects NativeWind availability and falls back to StyleSheet if needed.
 *
 * @example
 * ```tsx
 * <WrapperSafeAreaView edges={['top', 'bottom']}>
 *   <String>Safe area content</String>
 * </WrapperSafeAreaView>
 * ```
 * @param className - Custom CSS classes for the wrapper (NativeWind)
 * @param fill - Whether to fill the wrapper
 * @param px - Padding for the wrapper
 * @param style - Style prop
 * @param edges - Safe area edges to apply
 * @param emulateUnlessSupported - Whether to emulate unless supported
 * @param children - Child components to render
 * @returns SafeAreaView or fallback to scroll component
 */
export function WrapperSafeAreaView(props: PropsWithChildren<WrapperSafeAreaViewPropsTw>) {
  classNamePropsHandler(props, 'WrapperSafeAreaView');
  const { children, className, fill, px, style, ...rest } = props;

  if (!SafeAreaView) {
    return (
      <Wrapper
        {...styleHandler({ defaultStyle: wrapperSv({ fill, px }), style })}
        {...classNameHandler(cn(wrapperTv({ fill, px }), className))}
        {...rest}
      >
        {children}
      </Wrapper>
    );
  }

  return (
    <SafeAreaView
      {...styleHandler({ defaultStyle: wrapperSv({ fill, px }), style })}
      {...classNameHandler(cn(wrapperTv({ fill, px }), className))}
      {...rest}
    >
      {children}
    </SafeAreaView>
  );
}

WrapperSafeAreaView.displayName = 'WrapperSafeAreaView';
