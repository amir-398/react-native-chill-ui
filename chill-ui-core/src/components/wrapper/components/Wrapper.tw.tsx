import type { WrapperPropsTw } from '@types';
import type { PropsWithChildren } from 'react';

import { cn } from '@utils';
import { BoxTw } from '@components/box';

import { wrapperTv } from '../styles/Wrapper.tw.styles';
import { WrapperSafeAreaView } from './WrapperSafeAreaView.tw';

/**
 * The `<Wrapper />` component provides a flexible container with default styling and SafeAreaView support.
 *
 *
 * <!-- STORYBOOK_IMPORT_START
 * ```tsx
 * import { Wrapper } from 'react-native-chill-ui';
 * ```
 * STORYBOOK_IMPORT_END -->
 *
 * @example
 * ```tsx
 * <Wrapper className="bg-gray-100 p-4">
 *   <String>Content</String>
 * </Wrapper>
 * ```
 *
 * @param className - Custom CSS classes for the wrapper (NativeWind)
 * @param edges - Safe area edges to apply when hasSafeArea is true: `'top'` | `'right'` | `'bottom'` | `'left'`
 * @param fill - Whether to fill the wrapper
 * @param grow - Whether to grow the wrapper
 * @param hasSafeArea - Whether to wrap content in SafeAreaView
 * @param px - Horizontal padding variant: `'none'` | `'xs'` | `'sm'` | `'md'` | `'lg'` | `'xl'`
 * @param ViewProps - Any other props accepted by the native `View` component.
 */
export function Wrapper(props: PropsWithChildren<WrapperPropsTw>) {
  const { children, className, edges, fill, grow, hasSafeArea, px, ...rest } = props;

  const content = (
    <BoxTw className={cn(wrapperTv({ fill, grow, px }), className)} {...rest}>
      {children}
    </BoxTw>
  );

  if (hasSafeArea) {
    return (
      <WrapperSafeAreaView edges={edges} px="none">
        {content}
      </WrapperSafeAreaView>
    );
  }

  return content;
}

Wrapper.displayName = 'Wrapper';
