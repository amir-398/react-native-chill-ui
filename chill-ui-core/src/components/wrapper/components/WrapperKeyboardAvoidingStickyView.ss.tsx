import type { PropsWithChildren } from 'react';
import type { WrapperKeyboardAvoidingStickyViewPropsSs } from '@types';

import { customConsole } from '@utils';

import { Wrapper } from './Wrapper.ss';
import { wrapperSv } from '../styles/Wrapper.ss.styles';
import { WrapperSafeAreaView } from './WrapperSafeAreaView.ss';

// Optional import with error handling
let KeyboardStickyView: any;

try {
  // eslint-disable-next-line
  const keyboardController = require('react-native-keyboard-controller');
  if (keyboardController) {
    KeyboardStickyView = keyboardController.KeyboardStickyView;
  }
} catch {
  customConsole.warn(
    'react-native-keyboard-controller is not installed. To use WrapperKeyboardAvoidingView, please install it: npm install react-native-keyboard-controller',
  );
}

/**
 * KeyboardAvoidingView wrapper component for keyboard avoidance.
 *
 * @example
 * ```tsx
 * <WrapperKeyboardAvoidingStickyView behavior="padding">
 *   <Input placeholder="Type here" />
 * </WrapperKeyboardAvoidingStickyView>
 *
 * // With SafeAreaView
 * <WrapperKeyboardAvoidingStickyView hasSafeArea edges={['top', 'bottom']}>
 *   <Input placeholder="Type here" />
 * </WrapperKeyboardAvoidingStickyView>
 * ```
 * @param fill - Whether to fill the wrapper
 * @param grow - Whether to grow the wrapper
 * @param px - Padding for the wrapper
 * @param hasSafeArea - Whether to wrap content in SafeAreaView
 * @param edges - Safe area edges to apply when hasSafeArea is true
 * @param style - Style prop
 * @param enabled - Whether the keyboard avoiding view is enabled
 * @param offset - Offset for the keyboard avoiding sticky view { close: number; open: number }
 * @param children - Child components to render
 */
export function WrapperKeyboardAvoidingStickyView(props: PropsWithChildren<WrapperKeyboardAvoidingStickyViewPropsSs>) {
  const { children, edges, fill, grow, hasSafeArea, offset, px, style, ...rest } = props;

  if (!KeyboardStickyView) {
    return (
      <Wrapper style={[wrapperSv({ fill, grow, px }), style]} hasSafeArea={hasSafeArea} edges={edges} {...rest}>
        {children}
      </Wrapper>
    );
  }

  const content = (
    <KeyboardStickyView style={[wrapperSv({ fill, grow, px }), style]} offset={offset} {...rest}>
      {children}
    </KeyboardStickyView>
  );

  if (hasSafeArea) {
    return (
      <WrapperSafeAreaView edges={edges} px="none">
        {content}
      </WrapperSafeAreaView>
    );
  }

  return content;
}

WrapperKeyboardAvoidingStickyView.displayName = 'WrapperKeyboardAvoidingStickyView';
