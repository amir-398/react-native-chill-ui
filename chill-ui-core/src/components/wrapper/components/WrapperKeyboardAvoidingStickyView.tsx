import type { PropsWithChildren } from 'react';
import type { WrapperKeyboardAvoidingStickyViewPropsTw } from '@types';

import { classNameHandler, cn, customConsole, styleHandler } from '@utils';

import { Wrapper } from './Wrapper.tw';
import { wrapperTv } from '../styles/Wrapper.tw.styles';
import { wrapperSv } from '../styles/Wrapper.ss.styles';
import { WrapperSafeAreaView } from './WrapperSafeAreaView.tw';

// Optional import with error handling
let KeyboardStickyView: any;

try {
  // eslint-disable-next-line
  const keyboardController = require('react-native-keyboard-controller');
  if (keyboardController) {
    KeyboardStickyView = keyboardController.KeyboardStickyView;
  }
} catch {
  customConsole.warn(
    'react-native-keyboard-controller is not installed. To use WrapperKeyboardAvoidingView, please install it: npm install react-native-keyboard-controller',
  );
}

/**
 * The `<WrapperKeyboardAvoidingStickyView />` component provides a KeyboardStickyView wrapper for keyboard avoidance with sticky behavior.
 *
 *
 * <!-- STORYBOOK_IMPORT_START
 * ```tsx
 * import { WrapperKeyboardAvoidingStickyView } from 'react-native-chill-ui';
 * ```
 * STORYBOOK_IMPORT_END -->
 *
 * @example
 * ```tsx
 * <WrapperKeyboardAvoidingStickyView offset={{ close: 0, open: 20 }}>
 *   <Input placeholder="Type here" />
 * </WrapperKeyboardAvoidingStickyView>
 * ```
 *
 * @param className - Custom CSS classes for the wrapper (NativeWind only)
 * @param edges - Safe area edges to apply when hasSafeArea is true: `'top'` | `'right'` | `'bottom'` | `'left'`
 * @param enabled - Whether the keyboard avoiding sticky view is enabled (default: `true`)
 * @param fill - Whether to fill the wrapper
 * @param grow - Whether to grow the wrapper
 * @param hasSafeArea - Whether to wrap content in SafeAreaView
 * @param offset - Offset for the keyboard avoiding sticky view: `{ close: number; open: number }`
 * @param px - Horizontal padding variant: `'none'` | `'xs'` | `'sm'` | `'md'` | `'lg'` | `'xl'`
 * @param ViewProps - Any other props accepted by the native `View` component.
 */
export function WrapperKeyboardAvoidingStickyView(props: PropsWithChildren<WrapperKeyboardAvoidingStickyViewPropsTw>) {
  const { children, className, edges, fill, grow, hasSafeArea, px, style, ...rest } = props;

  if (!KeyboardStickyView) {
    return (
      <Wrapper
        {...styleHandler({ defaultStyle: wrapperSv({ fill, grow, px }), style })}
        {...classNameHandler(cn(wrapperTv({ fill, grow, px }), className))}
        hasSafeArea={hasSafeArea}
        edges={edges}
        {...rest}
      >
        {children}
      </Wrapper>
    );
  }

  const content = (
    <KeyboardStickyView
      {...styleHandler({ defaultStyle: wrapperSv({ fill, grow, px }), style })}
      {...classNameHandler(cn(wrapperTv({ fill, grow, px }), className))}
      {...rest}
    >
      {children}
    </KeyboardStickyView>
  );

  if (hasSafeArea) {
    return (
      <WrapperSafeAreaView edges={edges} px="none">
        {content}
      </WrapperSafeAreaView>
    );
  }

  return content;
}

WrapperKeyboardAvoidingStickyView.displayName = 'WrapperKeyboardAvoidingStickyView';
