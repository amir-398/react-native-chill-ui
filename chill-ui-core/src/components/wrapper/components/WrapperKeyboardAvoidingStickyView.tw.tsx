import type { PropsWithChildren } from 'react';
import type { WrapperKeyboardAvoidingStickyViewPropsTw } from '@types';

import { cn, customConsole } from '@utils';

import { Wrapper } from './Wrapper.tw';
import { wrapperTv } from '../styles/Wrapper.tw.styles';
import { WrapperSafeAreaView } from './WrapperSafeAreaView.tw';

// Optional import with error handling
let KeyboardStickyView: any;

try {
  // eslint-disable-next-line
  const keyboardController = require('react-native-keyboard-controller');
  if (keyboardController) {
    KeyboardStickyView = keyboardController.KeyboardStickyView;
  }
} catch {
  customConsole.warn(
    'react-native-keyboard-controller is not installed. To use WrapperKeyboardAvoidingView, please install it: npm install react-native-keyboard-controller',
  );
}

/**
 * KeyboardAvoidingView wrapper component for keyboard avoidance.
 *
 * @example
 * ```tsx
 * <WrapperKeyboardAvoidingView behavior="padding">
 *   <Input placeholder="Type here" />
 * </WrapperKeyboardAvoidingView>
 *
 * // With SafeAreaView
 * <WrapperKeyboardAvoidingView hasSafeArea edges={['top', 'bottom']}>
 *   <Input placeholder="Type here" />
 * </WrapperKeyboardAvoidingView>
 * ```
 * @param className - Custom CSS classes for the wrapper
 * @param fill - Whether to fill the wrapper
 * @param grow - Whether to grow the wrapper
 * @param px - Padding for the wrapper
 * @param hasSafeArea - Whether to wrap content in SafeAreaView
 * @param edges - Safe area edges to apply when hasSafeArea is true
 * @param style - Style prop
 * @param enabled - Whether the keyboard avoiding view is enabled
 * @param offset - Offset for the keyboard avoiding sticky view { close: number; open: number }
 * @param children - Child components to render
 */
export function WrapperKeyboardAvoidingStickyView(props: PropsWithChildren<WrapperKeyboardAvoidingStickyViewPropsTw>) {
  const { children, className, edges, fill, grow, hasSafeArea, px, style, ...rest } = props;

  if (!KeyboardStickyView) {
    return (
      <Wrapper
        className={cn(wrapperTv({ fill, grow, px }), className)}
        hasSafeArea={hasSafeArea}
        edges={edges}
        {...rest}
      >
        {children}
      </Wrapper>
    );
  }

  const content = (
    <KeyboardStickyView className={cn(wrapperTv({ fill, grow, px }), className)} {...rest}>
      {children}
    </KeyboardStickyView>
  );

  if (hasSafeArea) {
    return (
      <WrapperSafeAreaView edges={edges} px="none">
        {content}
      </WrapperSafeAreaView>
    );
  }

  return content;
}

WrapperKeyboardAvoidingStickyView.displayName = 'WrapperKeyboardAvoidingStickyView';
