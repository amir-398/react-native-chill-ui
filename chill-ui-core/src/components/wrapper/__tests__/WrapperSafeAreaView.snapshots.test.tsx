import React from 'react';
import { render } from '@testing-library/react-native';

import { WrapperSafeAreaView } from '../components/WrapperSafeAreaView';

describe('WrapperSafeAreaView Snapshots', () => {
  it('should match snapshot with basic props', () => {
    const tree = render(<WrapperSafeAreaView>Content</WrapperSafeAreaView>);
    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot with fill prop', () => {
    const tree = render(<WrapperSafeAreaView fill>Content</WrapperSafeAreaView>);
    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot with grow prop', () => {
    const tree = render(<WrapperSafeAreaView grow>Content</WrapperSafeAreaView>);
    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot with px prop', () => {
    const tree = render(<WrapperSafeAreaView px="md">Content</WrapperSafeAreaView>);
    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot with custom className', () => {
    const tree = render(<WrapperSafeAreaView className="custom-class">Content</WrapperSafeAreaView>);
    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot with custom style', () => {
    const tree = render(<WrapperSafeAreaView style={{ marginBottom: 8 }}>Content</WrapperSafeAreaView>);
    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot with edges prop', () => {
    const tree = render(<WrapperSafeAreaView edges={['top', 'bottom']}>Content</WrapperSafeAreaView>);
    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot with all edges', () => {
    const tree = render(<WrapperSafeAreaView edges={['top', 'left', 'right', 'bottom']}>Content</WrapperSafeAreaView>);
    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot with emulateUnlessSupported prop', () => {
    const tree = render(<WrapperSafeAreaView emulateUnlessSupported>Content</WrapperSafeAreaView>);
    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot with accessibility props', () => {
    const tree = render(
      <WrapperSafeAreaView accessible accessibilityLabel="Safe area container" testID="safe-area-view">
        Content
      </WrapperSafeAreaView>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('should match snapshot with all props combined', () => {
    const tree = render(
      <WrapperSafeAreaView
        fill
        grow
        px="lg"
        className="custom-class"
        style={{ marginBottom: 16 }}
        edges={['top', 'bottom']}
        emulateUnlessSupported
        testID="safe-area-view"
        accessible
        accessibilityLabel="Custom safe area view"
      >
        Content
      </WrapperSafeAreaView>,
    );
    expect(tree).toMatchSnapshot();
  });
});
