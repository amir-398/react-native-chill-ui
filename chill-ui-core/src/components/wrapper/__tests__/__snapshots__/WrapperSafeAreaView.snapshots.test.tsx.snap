// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WrapperSafeAreaView Snapshots should match snapshot with accessibility props 1`] = `
<RNCSafeAreaView
  accessibilityLabel="Safe area container"
  accessible={true}
  className="flex-1 px-[16px]"
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
  testID="safe-area-view"
>
  Content
</RNCSafeAreaView>
`;

exports[`WrapperSafeAreaView Snapshots should match snapshot with all edges 1`] = `
<RNCSafeAreaView
  className="flex-1 px-[16px]"
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
>
  Content
</RNCSafeAreaView>
`;

exports[`WrapperSafeAreaView Snapshots should match snapshot with all props combined 1`] = `
<RNCSafeAreaView
  accessibilityLabel="Custom safe area view"
  accessible={true}
  className="flex-grow px-[24px] flex-grow flex-none custom-class"
  edges={
    {
      "bottom": "additive",
      "left": "off",
      "right": "off",
      "top": "additive",
    }
  }
  emulateUnlessSupported={true}
  style={
    {
      "marginBottom": 16,
    }
  }
  testID="safe-area-view"
>
  Content
</RNCSafeAreaView>
`;

exports[`WrapperSafeAreaView Snapshots should match snapshot with basic props 1`] = `
<RNCSafeAreaView
  className="flex-1 px-[16px]"
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
>
  Content
</RNCSafeAreaView>
`;

exports[`WrapperSafeAreaView Snapshots should match snapshot with custom className 1`] = `
<RNCSafeAreaView
  className="flex-1 px-[16px] custom-class"
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
>
  Content
</RNCSafeAreaView>
`;

exports[`WrapperSafeAreaView Snapshots should match snapshot with custom style 1`] = `
<RNCSafeAreaView
  className="flex-1 px-[16px]"
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
  style={
    {
      "marginBottom": 8,
    }
  }
>
  Content
</RNCSafeAreaView>
`;

exports[`WrapperSafeAreaView Snapshots should match snapshot with edges prop 1`] = `
<RNCSafeAreaView
  className="flex-1 px-[16px]"
  edges={
    {
      "bottom": "additive",
      "left": "off",
      "right": "off",
      "top": "additive",
    }
  }
>
  Content
</RNCSafeAreaView>
`;

exports[`WrapperSafeAreaView Snapshots should match snapshot with emulateUnlessSupported prop 1`] = `
<RNCSafeAreaView
  className="flex-1 px-[16px]"
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
  emulateUnlessSupported={true}
>
  Content
</RNCSafeAreaView>
`;

exports[`WrapperSafeAreaView Snapshots should match snapshot with fill prop 1`] = `
<RNCSafeAreaView
  className="flex-1 px-[16px]"
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
>
  Content
</RNCSafeAreaView>
`;

exports[`WrapperSafeAreaView Snapshots should match snapshot with grow prop 1`] = `
<RNCSafeAreaView
  className="flex-grow px-[16px] flex-grow flex-none"
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
>
  Content
</RNCSafeAreaView>
`;

exports[`WrapperSafeAreaView Snapshots should match snapshot with px prop 1`] = `
<RNCSafeAreaView
  className="flex-1 px-[16px]"
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
>
  Content
</RNCSafeAreaView>
`;
